.github/workflows/unit_test_pipeline.yml

```diff
@@ -0,0 +1,66 @@
+name: TM1Py Integration Tests
+
+on:
+  workflow_dispatch:
+    inputs:
+      environments:
+        description: 'JSON array of environments to test (e.g., ["tm1-11-onprem", "tm1-11-cloud"])'
+        required: true
+        default: '["tm1-11-onprem","tm1-11-cloud","tm1-12-mcsp"]'
```

Comment:
It would be great if the pipeline is triggered when a new pull request is opened or when to an existing pull request (branch) is pushed.

Reply:
I agree 100%. The reason why I left this to be run on demand it's because it's new and people are not necessarily familiar with it. @MariusWirtz will most likely use it before merging anything against master, but not all contributors might want to do that nor might not know. 
I would suggest to leave it on demand BUT push for this to become the norm in the near future. Would you agree?

Reply:
Sold! Maybe for the future 😃